[TensorExpr] Add tensorexpr benchmarks.

ZolotukhinM opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack):
* #34234 [WIP] Changes in test/test_jit_fuser.py test/test_jit_fuser_te.py.
* #34233 [WIP] Changes to peephole.cpp.
* #34231 [WIP] Guard elimination changes from bertmaher/pytorch_fusion.
* **#34230 [TensorExpr] Add tensorexpr benchmarks.**
* #34228 [TensorExpr] Add LLVM codegen.
* #34227 [TensorExpr] Add CUDA codegen.
* #34226 [TensorExpr] Add a fuser pass based on tensor expressions.
* #34224 [TensorExpr] Pull changes to core classes for representing expressions and statements from the side branch.

This PR adds some benchmarks that we used to assess tensor expressions performance.

Differential Revision: [D20251830](https://our.internmc.facebook.com/intern/diff/D20251830)

Commit:
Add tensorexpr benchmarks.

[ghstack-poisoned]

Commit:
Update on "Add tensorexpr benchmarks."


Differential Revision: [D20251830](https://our.internmc.facebook.com/intern/diff/D20251830)

[ghstack-poisoned]

Commit:
Update descriptions. on "[TensorExpr] Add tensorexpr benchmarks."


This PR adds some benchmarks that we used to assess tensor expressions performance.

Differential Revision: [D20251830](https://our.internmc.facebook.com/intern/diff/D20251830)

[ghstack-poisoned]

Commit:
Lints on "[TensorExpr] Add tensorexpr benchmarks."


This PR adds some benchmarks that we used to assess tensor expressions performance.

Differential Revision: [D20251830](https://our.internmc.facebook.com/intern/diff/D20251830)

[ghstack-poisoned]

Commit:
Update on "[TensorExpr] Add tensorexpr benchmarks."


This PR adds some benchmarks that we used to assess tensor expressions performance.

Differential Revision: [D20251830](https://our.internmc.facebook.com/intern/diff/D20251830)

[ghstack-poisoned]

Commit:
Update on "[TensorExpr] Add tensorexpr benchmarks."


This PR adds some benchmarks that we used to assess tensor expressions performance.

Differential Revision: [D20251830](https://our.internmc.facebook.com/intern/diff/D20251830)

[ghstack-poisoned]

Commit:
Update on "[TensorExpr] Add tensorexpr benchmarks."


This PR adds some benchmarks that we used to assess tensor expressions performance.

Differential Revision: [D20251830](https://our.internmc.facebook.com/intern/diff/D20251830)

[ghstack-poisoned]

Commit:
Update on "[TensorExpr] Add tensorexpr benchmarks."


This PR adds some benchmarks that we used to assess tensor expressions performance.

Differential Revision: [D20251830](https://our.internmc.facebook.com/intern/diff/D20251830)

[ghstack-poisoned]

Commit:
Update on "[TensorExpr] Add tensorexpr benchmarks."


This PR adds some benchmarks that we used to assess tensor expressions performance.

Differential Revision: [D20251830](https://our.internmc.facebook.com/intern/diff/D20251830)

[ghstack-poisoned]

Commit:
Update on "[TensorExpr] Add tensorexpr benchmarks."


This PR adds some benchmarks that we used to assess tensor expressions performance.

Differential Revision: [D20251830](https://our.internmc.facebook.com/intern/diff/D20251830)

[ghstack-poisoned]

Commit:
Update on "[TensorExpr] Add tensorexpr benchmarks."


This PR adds some benchmarks that we used to assess tensor expressions performance.

Differential Revision: [D20251830](https://our.internmc.facebook.com/intern/diff/D20251830)

[ghstack-poisoned]

Commit:
Update on "[TensorExpr] Add tensorexpr benchmarks."


This PR adds some benchmarks that we used to assess tensor expressions performance.

Differential Revision: [D20251830](https://our.internmc.facebook.com/intern/diff/D20251830)

[ghstack-poisoned]

Commit:
Rebase on "[TensorExpr] Add tensorexpr benchmarks."


This PR adds some benchmarks that we used to assess tensor expressions performance.

Differential Revision: [D20251830](https://our.internmc.facebook.com/intern/diff/D20251830)

[ghstack-poisoned]

Commit:
Update on "[TensorExpr] Add tensorexpr benchmarks."


This PR adds some benchmarks that we used to assess tensor expressions performance.

Differential Revision: [D20251830](https://our.internmc.facebook.com/intern/diff/D20251830)

[ghstack-poisoned]

Commit:
Update on "[TensorExpr] Add tensorexpr benchmarks."


This PR adds some benchmarks that we used to assess tensor expressions performance.

Differential Revision: [D20251830](https://our.internmc.facebook.com/intern/diff/D20251830)

[ghstack-poisoned]

Commit:
Rebase on "[TensorExpr] Add tensorexpr benchmarks."


This PR adds some benchmarks that we used to assess tensor expressions performance.

Differential Revision: [D20251830](https://our.internmc.facebook.com/intern/diff/D20251830)

[ghstack-poisoned]

Commit:
Rebase on "[TensorExpr] Add tensorexpr benchmarks."


This PR adds some benchmarks that we used to assess tensor expressions performance.

Differential Revision: [D20251830](https://our.internmc.facebook.com/intern/diff/D20251830)

[ghstack-poisoned]

bertmaher reviewed (approved):
Strictly speaking we don't *need* these, but they're kind of a hand way to get some operator-level benchmarks.  Note that they're not representative of anything in particular, they're often just to see if we can hit max throughput on some arbitrarily selected operators.

In any case, I reviewed them in side branch: http://github.com/bertmaher/pytorch/tree/pytorch_fusion/

Commit:
Rebase and try to fix windows build. on "[TensorExpr] Add tensorexpr benchmarks."


This PR adds some benchmarks that we used to assess tensor expressions performance.

Differential Revision: [D20251830](https://our.internmc.facebook.com/intern/diff/D20251830)

[ghstack-poisoned]

Commit:
Update on "[TensorExpr] Add tensorexpr benchmarks."


This PR adds some benchmarks that we used to assess tensor expressions performance.

Differential Revision: [D20251830](https://our.internmc.facebook.com/intern/diff/D20251830)

[ghstack-poisoned]

Commit:
Update on "[TensorExpr] Add tensorexpr benchmarks."


This PR adds some benchmarks that we used to assess tensor expressions performance.

Differential Revision: [D20251830](https://our.internmc.facebook.com/intern/diff/D20251830)

[ghstack-poisoned]

Commit:
Update on "[TensorExpr] Add tensorexpr benchmarks."


This PR adds some benchmarks that we used to assess tensor expressions performance.

Differential Revision: [D20251830](https://our.internmc.facebook.com/intern/diff/D20251830)

[ghstack-poisoned]

Commit:
Update on "[TensorExpr] Add tensorexpr benchmarks."


This PR adds some benchmarks that we used to assess tensor expressions performance.

Differential Revision: [D20251830](https://our.internmc.facebook.com/intern/diff/D20251830)

[ghstack-poisoned]

Commit:
LLVM cmake integration changes on "[TensorExpr] Add tensorexpr benchmarks."


This PR adds some benchmarks that we used to assess tensor expressions performance.

Differential Revision: [D20251830](https://our.internmc.facebook.com/intern/diff/D20251830)

[ghstack-poisoned]

Commit:
Update on "[TensorExpr] Add tensorexpr benchmarks."


This PR adds some benchmarks that we used to assess tensor expressions performance.

Differential Revision: [D20251830](https://our.internmc.facebook.com/intern/diff/D20251830)

[ghstack-poisoned]

Commit:
Update on "[TensorExpr] Add tensorexpr benchmarks."


This PR adds some benchmarks that we used to assess tensor expressions performance.

Differential Revision: [D20251830](https://our.internmc.facebook.com/intern/diff/D20251830)

[ghstack-poisoned]

Commit:
Update on "[TensorExpr] Add tensorexpr benchmarks."


This PR adds some benchmarks that we used to assess tensor expressions performance.

Differential Revision: [D20251830](https://our.internmc.facebook.com/intern/diff/D20251830)

[ghstack-poisoned]

Commit:
Update on "[TensorExpr] Add tensorexpr benchmarks."


This PR adds some benchmarks that we used to assess tensor expressions performance.

Differential Revision: [D20251830](https://our.internmc.facebook.com/intern/diff/D20251830)

[ghstack-poisoned]

Commit:
Update on "[TensorExpr] Add tensorexpr benchmarks."


This PR adds some benchmarks that we used to assess tensor expressions performance.

Differential Revision: [D20251830](https://our.internmc.facebook.com/intern/diff/D20251830)

[ghstack-poisoned]

Commit:
Rebase on "[TensorExpr] Add tensorexpr benchmarks."


This PR adds some benchmarks that we used to assess tensor expressions performance.

Differential Revision: [D20251830](https://our.internmc.facebook.com/intern/diff/D20251830)

[ghstack-poisoned]
